### DISCUSSION TITLE
How can I know babel first compile is finish in watch mode?

### DISCUSSION THREAD

**Discussion by saifeiLee** [babel/babel#12455] [Q&A] [answered]
I am developing a react-ts-mono-repo project. Ant my compile script is: 
`lerna exec --parallel 'babel -w --root-mode upward src --out-dir dist --source-maps --extensions .tsx,.ts --delete-dir-on-start --no-comments'`.
My problem is:**When all packages compiled success, console not show any message.** How can I know the first compiling is 
 finish?
Any suggestions will be welcome. Thank you guys~

**Comment by nicolo-ribaudo** [comment]
Which Babel version are you using? It should log something like `Successfully compiled 10 packages with Babel`.

**Reply by saifeiLee** [reply]
@babel/cli@7.12.8
@babel/core@7.12.9

**Comment by saifeiLee** [comment, marked as answer]
<img width="942" alt="截屏2020-12-08 上午9 01 19" src="https://user-images.githubusercontent.com/28710808/101423424-029f2980-3934-11eb-9429-946547e3b602.png">
Every sub-package compiled success has logged a message to console, but there is not any message after all pkgs compiled success. Can anyone provide some advice? Thanks very much!

**Reply by nicolo-ribaudo** [reply]
Oh I see. The problem is that you are running the `babel ...` command multiple times: every one of them notifies you when it's finished, but they are completely independent and cannot know about eachother.

The solution is either to run a single `babel ...` command rather than `lerna exec` (it might not be possible, if you want to have multiple `dist` folders), or to write a small wrapper around `lerna exec` that counts how many time it logs `Succesfully compiled ...` and logs "DONE" when it's equal to the number of packages.

**Reply by saifeiLee** [reply]
That sounds like a solution. Thank you very much.
